src/alto-ui/Form/Typeahead/Typeahead.js

```diff
@@ -116,7 +116,8 @@ const Typeahead = React.forwardRef(
 
     const selectedString = itemToString(selectedItem);
 
-    const valueToString = (isFocused || isMenuOpen) && search !== null ? search : selectedString;
+    const valueToString =
+      (isFocused || isMenuOpen) && search !== null ? search : selectedString || value;
```

Comment:
```suggestion
      (isFocused || isMenuOpen) && search !== null ? search : (selectedString || value);
```
Just for improving code understanding and reading